src/registry/AvatarRegistry.sol

```diff
@@ -0,0 +1,541 @@
+
+    event RegistryEthReceived(address indexed sender, uint256 value);
+
+    ////////////////////////////////////////////////////////////////////////////
+    // MODIFIERS
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks whether a call is from the governance.
+    modifier onlyGovernance() {
+        if (msg.sender != governance) {
+            revert NotGovernance(msg.sender);
+        }
+        _;
+    }
+
+    /// @notice Checks whether a call is from the keeper.
+    modifier onlyKeeper() {
+        if (msg.sender != address(CL_REGISTRY)) {
+            revert NotKeeper(msg.sender);
+        }
+        _;
+    }
+
+    constructor(address _governance) {
+        if (_governance == address(0)) {
+            revert ZeroAddress();
+        }
+        governance = _governance;
+    }
+
+    /// @dev Fallback function accepts Ether transactions.
+    receive() external payable {
+        emit RegistryEthReceived(msg.sender, msg.value);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Governance
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @dev It will initiate the upKeep job for monitoring avatars
+    /// @notice only callable via governance
+    /// @param gasLimit gas limit for the avatar monitoring upkeep task
+    function avatarMonitoring(uint256 gasLimit) external onlyGovernance {
+        if (gasLimit == 0) {
+            revert ZeroUintValue();
+        }
+
+        avatarMonitoringUpKeepId = _registerUpKeep(address(this), gasLimit, AVATAR_REGISTRY_NAME);
+
+        if (avatarMonitoringUpKeepId > 0) {
+            /// @dev give allowance of spending LINK funds
+            LINK.approve(address(CL_REGISTRY), type(uint256).max);
+        }
+    }
+
+    /// @dev Adds an avatar into the registry
+    /// @notice only callable via governance
+    /// @param avatarAddress contract address to be register as new avatar
+    /// @param name avatar's name
+    /// @param gasLimit amount of gas to provide the target contract when
+    /// performing upkeep
+    function addAvatar(address avatarAddress, string memory name, uint256 gasLimit) external onlyGovernance {
+        /// @dev sanity checks before adding a new avatar in storage
+        if (avatarAddress == address(0)) {
+            revert ZeroAddress();
+        }
+        if (avatarsInfo[avatarAddress].gasLimit != 0) {
+            revert AvatarAlreadyRegister(avatarAddress);
+        }
+        if (gasLimit == 0) {
+            revert ZeroUintValue();
+        }
+        if (bytes(name).length == 0) {
+            revert EmptyString();
+        }
+
+        _avatars.add(avatarAddress);
+        avatarsInfo[avatarAddress] =
+            AvatarInfo({name: name, gasLimit: gasLimit, status: AvatarStatus.TESTING, upKeepId: 0});
+
+        emit NewAvatar(avatarAddress, name, gasLimit, block.timestamp);
+    }
```

Comment:
Maybe register upkeep when adding an avatar

Reply:
decided to move to this method the registration process in b66c905721bebe4a752e19374ba8f71c2f9df976. given that it's only one-off action, left the cl keeper focus solely in the topups of unfunded jobs